factory/src/test/java/com/google/auto/factory/processor/AutoFactoryProcessorTest.java

```diff
@@ -434,6 +434,28 @@ public void defaultPackage() {
         .generatesSources(loadExpectedFile("expected/DefaultPackageFactory.java"));
   }
 
+  @Test
+  public void classWithCopyAnnotations() {
+    JavaFileObject file = JavaFileObjects.forResource("good/ClassWithCopyAnnotations.java");
```

Comment:
Yes, I know the pain of reference-compare tests and when it goes to 100s of files. Not yet a problem here :-)

Why not divide and conquer? Split testing at the `FactoryDescriptor`.
1. Suppose the `FactoryWriter` is already seen as tested reliably (or is tested separately).
Then other tests might get access to the `FactoryDescriptor` that is passed into the FactoryWriter. 
This `FactoryDescriptor` contains all information needed for tests, to make them less over specified (as a text compare is).
Then it would be easy to add specific tiny unit tests, testing exactly one thing/behavior.
2. Write the java source text in unit test directly (hopefully soon with text blocks support). So the test source and asserts are together. Perhaps evolve an api to compose the java source texts with less duplication.